build.gradle.kts

```diff
@@ -0,0 +1,372 @@
+        runtimeClasspath {
+            exclude(group = "org.slf4j")
+            exclude(group = "org.jetbrains.kotlin")
+            exclude(group = "org.jetbrains.kotlinx")
+            exclude(group = "software.amazon.awssdk", module = "netty-nio-client")
+        }
+    }
+}
+
+// Kotlin plugin seems to be bugging out when there are no kotlin sources
+configure(subprojects.filter { it.name != "telemetry-client" }) {
+    apply(plugin = "kotlin")
+
+    sourceSets {
+        getOrCreate("integrationTest") {
+            java.srcDir("it")
+        }
+    }
+}
+
+subprojects {
+    group = parent!!.group
+    version = parent!!.version
+
+    apply(plugin = "java")
+    apply(plugin = "idea")
+    apply(plugin = "com.adarshr.test-logger")
+
+    sourceSets {
+        main.get().java.srcDirs(findFolders(project, "src", ideVersion))
+        main.get().resources.srcDirs(findFolders(project, "resources", ideVersion))
+        test.get().java.srcDirs(findFolders(project, "tst", ideVersion))
+        test.get().resources.srcDirs(findFolders(project, "tst-resources", ideVersion))
+        getOrCreate("integrationTest") {
+            compileClasspath += main.get().output + test.get().output
+            runtimeClasspath += main.get().output + test.get().output
+            java.srcDirs(findFolders(project, "it", ideVersion))
+            resources.srcDirs(findFolders(project, "it-resources", ideVersion))
+        }
+    }
+
+    val testArtifacts by configurations.creating
+    configurations.getByName("integrationTestImplementation").apply {
+        extendsFrom(configurations.getByName("testImplementation"))
+    }
+    configurations.getByName("integrationTestRuntimeOnly").apply {
+        extendsFrom(configurations.getByName("testRuntimeOnly"))
+    }
+
+    dependencies {
+        implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion")
+        implementation("org.jetbrains.kotlin:kotlin-reflect:$kotlinVersion")
+        testImplementation("com.nhaarman.mockitokotlin2:mockito-kotlin:$mockitoKotlinVersion")
+        testImplementation("org.mockito:mockito-core:$mockitoVersion")
+        testImplementation("org.assertj:assertj-core:$assertjVersion")
+        testImplementation("junit:junit:$junitVersion")
+    }
+
+    plugins.withType<TestLoggerPlugin> {
+        configure<TestLoggerExtension> {
+            showFullStackTraces = true
+            showStandardStreams = true
+            showPassedStandardStreams = false
+            showSkippedStandardStreams = true
+            showFailedStandardStreams = true
+        }
+    }
+
+    tasks.test {
+        configure<JacocoTaskExtension> {
+            // don't instrument sdk, icons, ktlint, etc.
+            includes = listOf("software.aws.toolkits.*")
+            excludes = listOf("software.aws.toolkits.ktlint.*")
+        }
+
+        reports {
+            junitXml.isEnabled = true
+            html.isEnabled = true
+        }
+    }
+
+    plugins.withType<IdeaPlugin> {
```

Comment:
Is this still needed with latest IDE versions + Gradle?

Reply:
Maybe, I don't want this to turn into the change everything about our build PR though. We can make further improvements later.